projects/core/src/lib/components/toolbar/geolocation/geolocation.component.ts

```diff
@@ -0,0 +1,127 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { MapService, CoordinateHelper } from '@tailormap-viewer/map';
+import { Subject, takeUntil, take } from 'rxjs';
+import { FeatureModel } from '@tailormap-viewer/api';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'tm-geolocation',
+  templateUrl: './geolocation.component.html',
+  styleUrls: ['./geolocation.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class GeolocationComponent implements OnInit, OnDestroy {
+  private destroyed = new Subject();
+  private featureGeom = new Subject<FeatureModel[]>();
+
+  private activeWatch: number | undefined = undefined;
+  private activeWatchTimeout = -1;
+
+  private static GEOLOCATION_TIMEOUT_MS = 12 * 1000;
+
+  public hasGeolocation = navigator.geolocation !== undefined;
+  public isWatching = false;
+  public hasFix = false;
+
+  constructor(
+    private mapService: MapService,
+    private snackBar: MatSnackBar,
+    private cdr: ChangeDetectorRef,
+  ) { }
+
+  public ngOnInit(): void {
+    this.mapService.renderFeatures$('geolocation-layer', this.featureGeom.asObservable(), f => {
+      if (f.__fid === 'geolocation-point') {
+        return {
+          styleKey: 'geolocation-point-style',
+          zIndex: 1000,
+          pointType: 'circle',
+          pointSize: 7,
+          pointFillColor: '#6236ff',
+          pointStrokeColor: '#ffffff',
+          pointStrokeWidth: 2,
+        };
+      } else {
+        return {
+          styleKey: 'geolocation-shadow-style',
+          zIndex: 999,
+          fillColor: '#6236ff',
+          fillOpacity: 20,
+          strokeColor: '#6236ff',
+          strokeOpacity: 40,
+        };
+      }
+     }).pipe(takeUntil(this.destroyed)).subscribe();
+  }
+
+  public ngOnDestroy(): void {
+    this.destroyed.next(null);
+    this.destroyed.complete();
+  }
+
+  private positionSuccess(pos: GeolocationPosition) {
+    this.mapService.getProjectionCode$()
+      .pipe(take(1))
+      .subscribe(projectionCode => {
+          const point = CoordinateHelper.projectCoordinates([ pos.coords.longitude, pos.coords.latitude ], 'EPSG:4326', projectionCode);
+          if (point === undefined) {
+              return;
+          }
+
+          const pointWkt = `POINT(${point[0]} ${point[1]})`;
+          const circleWkt = CoordinateHelper.circleFromWGS84CoordinatesAndRadius([ pos.coords.longitude, pos.coords.latitude ], pos.coords.accuracy, projectionCode);
```

Comment:
Why not 
```
const circleWkt = `CIRCLE(${point[0]} ${point[1]}, ${pos.coords.accuracy})`
```

Reply:
I agree the story is rewritten. This matches the story now

Reply:
the accuracy from the Geolocation API is in meters, but the radius of the OpenLayers(/custom WKT) circle is in degrees. And as a meter -> degree calculation can't be reliably done without its position on the sphere, this uses the same mechanism that OpenLayers uses for its internal geolocation support, which is [a circular polygon](https://openlayers.org/en/latest/apidoc/module-ol_geom_Polygon.html#.circular).